apps/reducers/lib/services/domoget.ex

```diff
@@ -53,6 +56,174 @@ defmodule Service.Domo do
+
+
+  def consolidate_shared_stores(dataset, type) do
+    dataset
+    |> merge_two_stores(type, "19", "297")
+    |> merge_two_stores(type, "59", "298")
+    |> merge_three_stores(type, "88", "299", "96")
+  end
+
+  def filter_data(dataset, type) do
+    Enum.map(dataset, fn{store, data} ->
+      {store, data_prep(data, String.to_existing_atom(type))} end
+      )
+  end
+
+
+
+    def data_prep(meta, :bin_audit)do
+      meta
+      |> remove_nulls("DATE")
+      |> filter_on_date("DATE")
+      |> sort_by_time_period("DATE")
+    end
+
+    def data_prep(meta, :challenge)do
+      meta
+      |> filter_on_date("start_date")
+      |> sort_by_time_period("start_date")
+    end
+
+    def data_prep(meta, :actuals)do
+      meta
+      |> remove_nulls("Count")
+      |> Enum.reject(fn (row) -> String.contains?(row["Count"], "#")  end)
+      |> filter_on_date("Week")
+      |> sort_by_time_period("Week")
+    end
+
+    def remove_nulls(data_list, field_name) do
+      Enum.reject(data_list, fn (row) -> row[field_name]== "\\N" end)
+    end
+
+    def filter_on_date(data_list, date_field_name)do
+      time_periods = Application.get_env(:reducers, Application.get_env(:reducers, :current_periods))
+      {_start_name, start_times} = hd(time_periods)
+      start_time = start_times.start_time
+
+      {_end_name, end_times} = List.last(time_periods)
+      end_time = end_times.end_time
+
+      Enum.filter(data_list, fn (row) -> date(row[date_field_name]) > start_time end)
+      |> Enum.filter(fn (row) -> date(row[date_field_name]) < end_time end)
+    end
+
+
+
+
+    def sort_by_time_period(filtered_meta, date_field) do
+      time_periods = Application.get_env(:reducers, Application.get_env(:reducers, :current_periods))
+      filtered_meta
+      |> Enum.group_by(fn(row)-> find_period(row, time_periods, date_field) end)
+    end
+
+    def find_period(row, season_periods, date_field) do
+      timestamp = date(row[date_field])
+      {period, _data} = season_periods
+      |>  Enum.find({:out_of_season, "data"}, fn {_period, %{:start_time => start_time, :end_time => end_time}} -> timestamp >= start_time and timestamp <= end_time end)
+      to_string(period)
+    end
+
+
+  def get_timestamp({:bin_audit, event}), do: date(event.data["DATE"])
+  def get_timestamp({:actuals, event}), do: date(event.data["Week"])
+  def get_timestamp({:pre_actual, event}), do: date(event.data["Week"])
+  def get_timestamp({:refill_actual, event}), do: date(event.data["Week"])
+  def get_timestamp({:pre_challenge, event})do
+    max = Enum.max_by(event.data["users"], fn(user) -> elem(user, 1)["start_time"] end)
+    elem(max, 1)["start_time"]
+  end
+  def get_timestamp({:refill_challenge, event})do
+    max = Enum.max_by(event.data["users"], fn(user) -> elem(user, 1)["start_time"] end)
+    elem(max, 1)["start_time"]
+  end
+
+  def date(datestring) when datestring == "", do: 1494287000000 #this date will be before season1 starts, so it won't get played
```

Comment:
May want to comment what the job of the date function is.  I can see the pattern match on a specific date which represents tournament start time.

But when looking at just this set of functions by itself, I can't really tell what should be happening.

Any idea how much test coverage (% lines covered) the domoget module has?